src/main/java/core/basesyntax/ArrayList.java

```diff
@@ -1,48 +1,100 @@
 package core.basesyntax;
 
+import java.util.NoSuchElementException;
+
 public class ArrayList<T> implements List<T> {
+    private static final String INDEX_EXCEPTION_MESSAGE = "Index Out of Bound Exception";
+    private static final int INITIAL_CAPACITY = 10;
+    private int size;
+    private T[] array;
+
+    public ArrayList() {
+        this.array = (T[]) new Object[INITIAL_CAPACITY];
+    }
+
     @Override
     public void add(T value) {
-
+        provideSize();
+        array[size] = value;
+        size++;
     }
 
     @Override
     public void add(T value, int index) {
-
+        areIndexExist(index);
+        provideSize();
+        System.arraycopy(array, index, array, index + 1, size - index);
+        array[index] = value;
+        size++;
     }
 
     @Override
     public void addAll(List<T> list) {
-
+        for (int i = 0; i < list.size(); i++) {
+            add(list.get(i));
+        }
     }
 
     @Override
     public T get(int index) {
-        return null;
+        areIndexExist(index + 1);
+        return array[index];
     }
 
     @Override
     public void set(T value, int index) {
-
+        areIndexExist(index + 1);
+        array[index] = value;
     }
 
     @Override
     public T remove(int index) {
-        return null;
+        areIndexExist(index + 1);
+        T value = array[index];
+        System.arraycopy(array, index + 1, array, index, size - index - 1);
+        size--;
```

Comment:
what about element on that index? will he be stored in memory still?

Reply:
yes, but it should be removed